test/basic_json.cpp

```diff
@@ -192,6 +192,15 @@ TEST(general, map_to_root)
     EXPECT_EQ(m["bar"], 20);
 }
 
+TEST(general, json_stream_operator)
+{
+    const char *exp;
+    std::map<std::string, int> m({{"bar", 2}, {"foo", 1}});
+    Tree t;
+    t.rootref() << m;
+    std::cout << as_json(t) << std::endl;
```

Comment:
this is not a test. this is just coverage. a test requires EXPECTs and there are none here.

Reply:
send to a stringstream instead of cout, parse the output string and verify with calls to EXPECT that the trees compare equal.

Reply:
I would argue that it's still a test since it validates that the stream operator compiles successfully, but nevermind. I added some parsing and EXPECT calls :-)